### ISSUE TITLE
Provide a way to enable stacktrace recovery in Android release build

### ISSUE THREAD

**Issue by varahash** [Kotlin/kotlinx.coroutines#1998]
As mentioned in https://github.com/Kotlin/kotlinx.coroutines/blob/master/ui/kotlinx-coroutines-android/README.md#optimization

> When optimizations are enabled with R8 version 1.6.0 or later the following debugging features are permanently turned off to reduce the size of the resulting binary:
> 
> - Debugging mode
> - Stacktrace recovery
> - The internal assertions in the library are also permanently removed.

There is should be a way to opt-out this behaviour.

**Comment by Thomas-Vos** [comment]
Yes, please. I really need the stack trace of a `kotlinx.coroutines.JobCancellationException` so I can try to reproduce the reported crash. Is there a way to enable stack trace recovery in release builds?

**Comment by dkhalanskyjb** [comment]
See #3611 for some prior art.

**Comment by LouisCAD** [comment]
What is causing stacktrace recovery to not work after R8?
Is it rules that the kotlinx.coroutines library is embedding?

**Comment by qwwdfsad** [comment]
Yes, embedded rules are the main suspect here

**Comment by LouisCAD** [comment]
I mean, just using Gradle APIs should suffice to exclude it if it's in a separate artifact.

**Comment by Anamorphosee** [comment]
FYI [Decoroutinator](https://github.com/Anamorphosee/stacktrace-decoroutinator) works for Android release build and [is used by some applications](https://github.com/square/retrofit/issues/3474#issuecomment-3126169260) in production.
